### ISSUE TITLE
A friend connecting sometimes crashes my entire internet until PC restart

### ISSUE THREAD

**Issue by Jarelk** [bluemagic123/tModLoader#3194]
### Version

Latest 1.4-stable

### OS

Windows

### Platform

Steam

### This bug affects

Gameplay as a Player

### Description

Sometimes, when my friend connects to my tmodloader (with Thorium) server, he manages to crash my internet completely. It's so bad that I can't soft-shutdown my PC, and I have to hold power button or reset.

Server settings are not using steam server, and not automatically port forwarding. The server and client logs don't seem to show anything notworthy happening either, besides disconnections.

### Steps to reproduce

1. Host a server via the .bat file, on port 7777. I port forwarded (on the crappy Genexis HRG1000 Platinum-4810 we get in our housing unit). No steam server, no automatic port forwarding.
2. A friend joins the game. Roughly 20% of the time, it causes the issue.

### Expected Behavior

My internet should **not** die.

### Actual Behavior

My internet very much dies. When I check network adapter config, Ethernet becomes grayed out. Might indicate some sort of driver crash?

### Reproduction frequency

Roughly 20% of the time he tries to join.

### Additional Information

I'm running on Windows 11, and use ESET Internet Security. My router is the Genexis HRG1000 Platinum-4810. 

This is unlike any issue I've ever had with any kind of server hosting, so I'm kind of stumped to be honest.

**Comment by Kogsey** [comment]
My dad actually started having the same issues with another game Hero's and Generals, so this could be one of those funny mixes of windows or drivers spitting it on some weird exception, or they could be unrelated, just spitballing. I'm assuming if tmodloader has no error it's a driver crash, could you poke in device manager for its crash report?

**Comment by direwolf420** [comment]
Make sure both of you have the exact same versions of mods enabled. Large mods can "choke" a bad internet connection because it will download the matching version via p2p.

**Comment by pudquick** [comment]
I can confirm the exact same issue.

Can confirm that the mod versions and tModLoader versions, and Terraria client and server versions are identical across all of the devices in question.

**Comment by pudquick** [comment]
I tried moving it to a Docker container on Windows. With with WSL2 backend, instead of locking up the internet it eventually just panics the entire computer. Going to move it to Hyper-V backend instead to more properly insulate the server from the OS and see if that makes a difference.

**Comment by KaptenLada** [comment]
I had the same problem but managed to fix it. I think it will work for you guys too! It's not a port problem because I opened all ports and disabled firewall and tried. 

Then I tried to update the driver on the network adapter and it worked. Here is the driver that I installed from[ intels site](https://www.intel.com/content/www/us/en/download/15084/intel-ethernet-adapter-complete-driver-pack.html).

**Comment by JavidPack** [comment]
This issue has either no recent activity or no reproducible steps, so it is being closed. We assume this issue has solved itself in some manner. Feel free to open a new issue if the issue persists and fill out the issue template completely.

**Comment by velcz** [comment]
This is an old thread, but I wish to leave this here for anyone who runs into this same niche issue as I did and ends up here while googling to find help. I managed to solve this issue on my computer, which it turned out to be caused by sh***y unstable network hardware on my MSI motherboard that TModloader easily overloaded with the massive amount of data it sends when stacking up mods. 

### The fix that worked for me was the following:

Press Windows + X and go to Device Manager, then Network adapters.

Right click your ethernet adapter, and go to Properties, the the Advanced Tab.

Here, you want to set the following properties to "Disabled":

- Flow Control
- Interrupt Moderation
- IPv4 Checksum Offload
- Large Send Offload (IPv4)
- TCP & UDP Checksum Offload
- Any other "power saving"-esque settings, just turn them off

I also went in the Power Management tab and turned off "Allow the computer to turn off this device to save power"

Basically what this is doing is telling your ethernet connection to just ignore offloading net processing to the network hardware on your motherboard and letting your actual processor handle the load, with no restrictions to be more "efficient". **Not only did this stop my internet from crashing, but we are in the middle of our most lag-free playthrough we've ever had in TModLoader.**

Hope that this finds at least one person who needs to see it. Cheers.
